### PR TITLE
DOC: Clarify that astropy.logger is not for everyone

### LINKED ISSUE #11500: Warnings logging doesn't log warnings
### Description

The Astropy logging mechanism provides the `enable_warnings_logging()` function, which is claimed to enable logging of `warnings.warn()` messages. This does not appear to work.

### Expected behavior

Warnings signaled by `warnings.warn` should be recorded in the log.

### Actual behavior

Such warnings do not appear.

### Steps to Reproduce

```python
import warnings
from astropy import log
import astropy.logger
import logging

try:
    log.enable_warnings_logging()
except astropy.logger.LoggingError:
    pass

with log.log_to_file("test.log"):
    log.warning("logger warning")
    warnings.warn("warnings warning")
```

The file `test.log` should contain both "logger warning" and "warnings warning" but it only contains "logger warning".

(The exception handling is because the only way to ensure the astropy warnings logging mechanism is turned on is to call the function and catch the exception if it's already turned on; it is in my completely default setup.)

### System Details

```
>>> import platform; print(platform.platform())
Linux-3.10.0-862.14.4.el7.x86_64-x86_64-with-debian-buster-sid
>>> import sys; print("Python", sys.version)
Python 3.7.6 | packaged by conda-forge | (default, Jun  1 2020, 18:57:50)
[GCC 7.5.0]
>>> import numpy; print("Numpy", numpy.__version__)
Numpy 1.18.5
>>> import astropy; print("astropy", astropy.__version__)
astropy 4.0
>>> import scipy; print("Scipy", scipy.__version__)
Scipy 1.4.1
>>> import matplotlib; print("Matplotlib", matplotlib.__version__)
Matplotlib 3.0.3
```

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to clarify the scope of `astropy.logger`. cc @aarchiba 

Attempt to address #11500 , close #11501 

p.s. If you think this needs to be backported, feel free to re-milestone.

### CODE REVIEW

File: astropy/logger.py
Comment (adrn) [inline astropy/logger.py:7]: Maybe we want to be more explicit? For example,
```suggestion
    This module is meant for internal ``astropy`` usage.
    For use in other packages, we recommend implementing your own logger class
    and possibly using this implementation as a template.
```

Comment (saimn) [reply astropy/logger.py:7]: I would not people make a class, because then you need to set the logger class before instantiating a logger (https://github.com/astropy/astropy/blob/838d92e4bc03d9ca992ed327b986d87890106413/astropy/logger.py#L97-L103), which then leads to instantiating a global logger because it's pain. logging is not meant to be used like this.

Comment (pllim) [reply astropy/logger.py:7]: Is the suggested text okay if I remove "class" from the words?

Comment (saimn) [reply astropy/logger.py:7]: Yeah, I would also remove the end of the sentence ;)

Comment (pllim) [reply astropy/logger.py]: Okay, I updated the text. What about now?

Comment (saimn) [reply astropy/logger.py]: Yes, works for me !

**Comment by hamogu** [comment]
@aarchiba Would this doc change have solved the problem in #11500, in other words, would you not have opened #11500 if the docs suggested here are in place?

@pllim Don't get me wrong: I think this change should be merged. I'm just not sure if it really fixes #11500 or if #11500 should still be open even after this is merged.

**Comment by pllim** [comment]
@hamogu , I don't see us further changing the logging internals regardless. I will just close  #11500 as won't fix even if this PR doesn't close it.

@adrn or @saimn , can I get an approval or is there more change needed? Thanks!

**Review by saimn — APPROVED** [review]
Sounds good to me.

**Pull request merged**
